cmd/kube-proxy/app/server.go

```diff
@@ -436,6 +474,9 @@ with the apiserver API to configure the proxy.`,
 			if err := opts.Complete(); err != nil {
 				klog.Fatalf("failed complete: %v", err)
 			}
+			if err := opts.ValidateMixedArguments(cmd.Flags()); err != nil {
+				klog.Exit(err)
```

Comment:
Should we log a warning message instead of exiting? Making this fatal may break existing user who starts using --config but forgets to remove the legacy flags.

Reply:
I would recommend preventing running in mixed mode. A warning will be lost in log noise.

> Making this fatal may break existing user who starts using --config but forgets to remove the legacy flags.

Preventing them from running a confused invocation is exactly the goal.